From 923e1a9142f88b4970a3a860b6e41bf74dea4091 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 19 May 2021 17:56:03 -0700
Subject: [PATCH] clang: Fix x86 triple for non-debian multiarch linux distros

OpenEmbedded does not hardcode mutli-arch like debian therefore ensure
that it still uses the proper tuple

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 clang/lib/Driver/ToolChains/Linux.cpp | 3 +++
 1 file changed, 3 insertions(+)

--- a/clang/lib/Driver/ToolChains/Linux.cpp
+++ b/clang/lib/Driver/ToolChains/Linux.cpp
@@ -622,6 +622,9 @@ void Linux::addLibStdCxxIncludePaths(con
       GCCInstallation.getTriple().getArch() == llvm::Triple::x86
           ? "i386-linux-gnu"
           : TripleStr;
+  // OpenEmbedded does not hardcode the triple to i386-linux-gnu like debian
+  if (GCCInstallation.getTriple().getVendor() == llvm::Triple::OpenEmbedded)
+	  DebianMultiarch = TripleStr;
 
   // Try generic GCC detection first.
   if (Generic_GCC::addGCCLibStdCxxIncludePaths(DriverArgs, CC1Args,
